File: casadaApi/src/main/java/com/casada/casadaApi/entities/Orderproduct.java
package com.casada.casadaApi.entities;

import lombok.Data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serializable;

/**
 * $table.getTableComment()
 */
@Data
@Entity
@Table(name = "orderproduct")
public class Orderproduct implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "orderid", nullable = false)
    private Long orderid;

    @Id
    @Column(name = "productid", nullable = false)
    private Long productid;

    @Column(name = "discount", nullable = false)
    private Double discount;

    @Column(name = "guarantee", nullable = false)
    private Long guarantee;

    @Column(name = "quantity", nullable = false)
    private Long quantity;

}

File: casadaGUI/casada/lib/data/order_product.dart
class OrderProduct {
  int? orderid;
  int? productid;
  String? productName;
  double? productPrice;
  double? discount;
  int? guarantee;
  int? quantity;

  OrderProduct(
      {this.orderid,
      this.productid,
      this.productName,
      this.productPrice,
      this.discount,
      this.guarantee,
      this.quantity});

  OrderProduct.fromJson(Map<String, dynamic> json) {
    orderid = json['orderid'];
    productid = json['productid'];
    productName = json['productName'];
    productPrice = json['productPrice'];
    discount = json['discount'];
    guarantee = json['guarantee'];
    quantity = json['quantity'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['orderid'] = this.orderid;
    data['productid'] = this.productid;
    data['productName'] = this.productName;
    data['productPrice'] = this.productPrice;
    data['discount'] = this.discount;
    data['guarantee'] = this.guarantee;
    data['quantity'] = this.quantity;
    return data;
  }
}